tests/test_django/urls.py

```
@@ -52,6 +119,83 @@
         return Q()
 
 
+class GatedAuthorCrudl(CrudlController[Author], DefaultFilter[Author]):  # pylint: disable=too-many-ancestors
+    """CRUDL controller for the Author model, with permission gating."""
+
+    @override
+    def post_create(self, request: RequestDetails[TDjangoModel]) -> None:
+        """Pre-create hook to check permissions before creating an author."""
+        if not request.object:
+            raise ValueError("Created object is not set.")
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)